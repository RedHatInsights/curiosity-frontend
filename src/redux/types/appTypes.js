/**
 * @memberof Types
 * @module AppTypes
 */

const STATUS_4XX = '4XX';
const STATUS_5XX = '5XX';
const SET_PRODUCT = 'SET_PRODUCT';
const SET_PRODUCT_VARIANT = 'SET_PRODUCT_VARIANT';
const SET_PRODUCT_VARIANT_QUERY_RESET_ALL = 'SET_PRODUCT_VARIANT_QUERY_RESET_ALL';

/**
 * Application action, reducer types.
 *
 * @type {{STATUS_4XX: string, SET_PRODUCT_VARIANT: string, SET_PRODUCT_VARIANT_QUERY_RESET_ALL: string,
 *     SET_PRODUCT: string, STATUS_5XX: string}}
 */
const appTypes = {
  STATUS_4XX,
  STATUS_5XX,
  SET_PRODUCT,
  SET_PRODUCT_VARIANT,
  SET_PRODUCT_VARIANT_QUERY_RESET_ALL
};

export {
  appTypes as default,
  appTypes,
  STATUS_4XX,
  STATUS_5XX,
  SET_PRODUCT,
  SET_PRODUCT_VARIANT,
  SET_PRODUCT_VARIANT_QUERY_RESET_ALL
};
