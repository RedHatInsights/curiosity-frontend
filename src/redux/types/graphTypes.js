/**
 * @memberof Types
 * @module GraphTypes
 */

const SET_GRAPH_LEGEND = 'SET_GRAPH_LEGEND';

/**
 * Graph action, reducer types.
 *
 * @type {{SET_GRAPH_LEGEND: string}}
 */
const graphTypes = {
  SET_GRAPH_LEGEND
};

export { graphTypes as default, graphTypes, SET_GRAPH_LEGEND };
