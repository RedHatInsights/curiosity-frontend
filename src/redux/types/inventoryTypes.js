const SET_INVENTORY_TAB = 'SET_INVENTORY_TAB';

/**
 * Inventory action, reducer types.
 *
 * @type {{SET_INVENTORY_TAB: string}}
 */
const inventoryTypes = {
  SET_INVENTORY_TAB
};

export { inventoryTypes as default, inventoryTypes, SET_INVENTORY_TAB };
