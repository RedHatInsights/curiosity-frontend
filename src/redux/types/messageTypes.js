/**
 * @memberof Types
 * @module MessageTypes
 */

const SET_BANNER_MESSAGES = 'SET_BANNER_MESSAGES';

/**
 * User action, reducer types.
 *
 * @type {{SET_BANNER_MESSAGES: string}}
 */
const messageTypes = { SET_BANNER_MESSAGES };

export { messageTypes as default, messageTypes, SET_BANNER_MESSAGES };
