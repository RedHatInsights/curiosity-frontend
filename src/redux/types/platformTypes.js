const PLATFORM_INIT = 'PLATFORM_INIT';
const PLATFORM_APP_NAME = 'PLATFORM_APP_NAME';
const PLATFORM_ON_NAV = 'PLATFORM_ON_NAV';
const PLATFORM_SET_NAV = 'PLATFORM_SET_NAV';

const platformTypes = { PLATFORM_INIT, PLATFORM_APP_NAME, PLATFORM_ON_NAV, PLATFORM_SET_NAV };

export { platformTypes as default, platformTypes, PLATFORM_INIT, PLATFORM_APP_NAME, PLATFORM_ON_NAV, PLATFORM_SET_NAV };
