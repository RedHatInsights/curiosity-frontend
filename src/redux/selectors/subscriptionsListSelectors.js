import { createSelectorCreator, defaultMemoize } from 'reselect';
import _isEqual from 'lodash/isEqual';
import { rhsmApiTypes } from '../../types/rhsmApiTypes';
import { reduxHelpers } from '../common/reduxHelpers';
import { apiQueries } from '../common';
import { selector as userSession } from './userSelectors';

/**
 * ToDo: Consider consolidating inventory selectors, and/or create API specific selectors, i.e. RHSM, etc
 * Breaking out the inventory selectors is a temporary solution until the API is finalized. Aspects
 * of the caching and applying the API schemas is now consistent enough to allow for grouping/refinement.
 */

/**
 * Create a custom "are objects equal" selector.
 *
 * @private
 * @type {Function}}
 */
const createDeepEqualSelector = createSelectorCreator(defaultMemoize, _isEqual);

/**
 * Selector cache.
 *
 * @private
 * @type {{dataId: {string}, data: {object}}}
 */
const selectorCache = { dataId: null, data: {} };

/**
 * Return a combined state, props object.
 *
 * @private
 * @param {object} state
 * @param {object} props
 * @returns {object}
 */
const statePropsFilter = (state, props = {}) => ({
  ...state.inventory?.subscriptionsInventory?.[props.productId],
  ...{
    viewId: props.viewId,
    productId: props.productId
  }
});

/**
 * Return a combined query object.
 *
 * @param {object} state
 * @param {object} props
 * @returns {object}
 */
const queryFilter = (state, props = {}) => {
  const { inventoryQuery: query } = apiQueries.parseRhsmQuery({
    ...props.query,
    ...state.view?.subscriptionsQuery?.[props.productId],
    ...state.view?.query?.[props.productId],
    ...state.view?.query?.[props.viewId]
  });

  return query;
};

/**
 * Create selector, transform combined state, props into a consumable object.
 *
 * @type {{pending: boolean, fulfilled: boolean, listData: object, error: boolean, status: (*|number)}}
 */
const selector = createDeepEqualSelector([statePropsFilter, queryFilter], (response, query = {}) => {
  const { viewId = null, productId = null, metaId, ...responseData } = response || {};

  const updatedResponseData = {
    error: responseData.error || false,
    fulfilled: false,
    pending: responseData.pending || responseData.cancelled || false,
    listData: [],
    itemCount: 0,
    query,
    status: responseData.status
  };

  const cache =
    (viewId && productId && selectorCache.data[`${viewId}_${productId}_${JSON.stringify(query)}`]) || undefined;

  Object.assign(updatedResponseData, { ...cache });

  // Reset cache on viewId update
  if (viewId && selectorCache.dataId !== viewId) {
    selectorCache.dataId = viewId;
    selectorCache.data = {};
  }

  /**
   * ToDo: evaluate removing id check
   * This logic is left over. We started cancelling subsequent promise calls making additional
   * checks superfluous, fulfilled should be the only check necessary now.
   */
  if (responseData.fulfilled && productId === metaId) {
    const {
      [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_DATA]: listData = [],
      [rhsmApiTypes.RHSM_API_RESPONSE_META]: listMeta = {}
    } = responseData.data || {};

    updatedResponseData.listData.length = 0;

    // Apply "display logic" then return a custom value for entries
    const customInventoryValue = ({ key, value }) => {
      switch (key) {
        case rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.UPCOMING_EVENT_DATE:
          return (value && new Date(value)) || null;
        default:
          return value ?? null;
      }
    };

    // Generate normalized properties
    const [updatedListData, updatedListMeta] = reduxHelpers.setNormalizedResponse(
      {
        schema: rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES,
        data: listData,
        customResponseValue: customInventoryValue
      },
      {
        schema: rhsmApiTypes.RHSM_API_RESPONSE_META_TYPES,
        data: listMeta
      }
    );

    const [meta = {}] = updatedListMeta || [];

    // Update response and cache
    updatedResponseData.itemCount = meta[rhsmApiTypes.RHSM_API_RESPONSE_META_TYPES.COUNT] ?? 0;
    updatedResponseData.listData = updatedListData;
    updatedResponseData.fulfilled = true;
    selectorCache.data[`${viewId}_${productId}_${JSON.stringify(query)}`] = {
      ...updatedResponseData
    };
  }

  return updatedResponseData;
});

/**
 * Expose selector instance. For scenarios where a selector is reused across component instances.
 *
 * @param {object} defaultProps
 * @returns {{pending: boolean, fulfilled: boolean, graphData: object, error: boolean, session: object,
 *     status: (*|number)}}
 */
const makeSelector = defaultProps => (state, props) => ({
  ...userSession(state, props, defaultProps),
  ...selector(state, props, defaultProps)
});

const subscriptionsListSelectors = {
  subscriptionsList: selector,
  makeSubscriptionsList: makeSelector
};

export { subscriptionsListSelectors as default, subscriptionsListSelectors, selector, makeSelector };
