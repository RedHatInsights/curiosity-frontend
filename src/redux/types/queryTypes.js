import { RHSM_API_QUERY_TYPES } from '../../types/rhsmApiTypes';

const SET_QUERY_CLEAR = 'SET_QUERY_CLEAR';
const SET_QUERY_CLEAR_INVENTORY_LIST = 'SET_QUERY_CLEAR_INVENTORY_LIST';

const SET_QUERY_RHSM_TYPES = {
  [RHSM_API_QUERY_TYPES.GRANULARITY]: `SET_QUERY_RHSM_${RHSM_API_QUERY_TYPES.GRANULARITY}`,
  [RHSM_API_QUERY_TYPES.SLA]: `SET_QUERY_RHSM_${RHSM_API_QUERY_TYPES.SLA}`,
  [RHSM_API_QUERY_TYPES.UOM]: `SET_QUERY_RHSM_${RHSM_API_QUERY_TYPES.UOM}`,
  [RHSM_API_QUERY_TYPES.USAGE]: `SET_QUERY_RHSM_${RHSM_API_QUERY_TYPES.USAGE}`
};

const SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES = {
  [RHSM_API_QUERY_TYPES.DIRECTION]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.DIRECTION}`,
  [RHSM_API_QUERY_TYPES.SORT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.SORT}`,
  [RHSM_API_QUERY_TYPES.LIMIT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.LIMIT}`,
  [RHSM_API_QUERY_TYPES.OFFSET]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.OFFSET}`
};

const SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_TYPES = {
  [RHSM_API_QUERY_TYPES.DIRECTION]: `SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_${RHSM_API_QUERY_TYPES.DIRECTION}`,
  [RHSM_API_QUERY_TYPES.SORT]: `SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_${RHSM_API_QUERY_TYPES.SORT}`,
  [RHSM_API_QUERY_TYPES.LIMIT]: `SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_${RHSM_API_QUERY_TYPES.LIMIT}`,
  [RHSM_API_QUERY_TYPES.OFFSET]: `SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_${RHSM_API_QUERY_TYPES.OFFSET}`
};

/**
 * Query/filter reducer types.
 *
 * @type {{SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_TYPES: string, SET_QUERY_RHSM_TYPES: string,
 *     SET_QUERY_CLEAR: string, SET_QUERY_CLEAR_INVENTORY_LIST: string,
 *     SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES: string}}
 */
const queryTypes = {
  SET_QUERY_CLEAR,
  SET_QUERY_CLEAR_INVENTORY_LIST,
  SET_QUERY_RHSM_TYPES,
  SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES,
  SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_TYPES
};

export {
  queryTypes as default,
  queryTypes,
  SET_QUERY_CLEAR,
  SET_QUERY_CLEAR_INVENTORY_LIST,
  SET_QUERY_RHSM_TYPES,
  SET_QUERY_RHSM_HOSTS_INVENTORY_TYPES,
  SET_QUERY_RHSM_SUBSCRIPTIONS_INVENTORY_TYPES
};
