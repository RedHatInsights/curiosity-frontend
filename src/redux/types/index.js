import { appTypes } from './appTypes';
import { graphTypes } from './graphTypes';
import { platformTypes } from './platformTypes';
import { queryTypes } from './queryTypes';
import { rhsmTypes } from './rhsmTypes';
import { toolbarTypes } from './toolbarTypes';
import { userTypes } from './userTypes';

const reduxTypes = {
  app: appTypes,
  graph: graphTypes,
  platform: platformTypes,
  query: queryTypes,
  rhsm: rhsmTypes,
  toolbar: toolbarTypes,
  user: userTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  appTypes,
  graphTypes,
  platformTypes,
  queryTypes,
  rhsmTypes,
  toolbarTypes,
  userTypes
};
