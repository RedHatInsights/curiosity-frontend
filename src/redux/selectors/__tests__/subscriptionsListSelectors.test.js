import subscriptionsListSelectors from '../subscriptionsListSelectors';
import { rhsmApiTypes } from '../../../types/rhsmApiTypes';

describe('SubscriptionsListSelectors', () => {
  it('should return specific selectors', () => {
    expect(subscriptionsListSelectors).toMatchSnapshot('selectors');
  });

  it('should pass minimal data on missing a reducer response', () => {
    const state = {};
    expect(subscriptionsListSelectors.subscriptionsList(state)).toMatchSnapshot('missing reducer error');
  });

  it('should pass minimal data on a product ID without a product ID provided', () => {
    const props = {
      viewId: 'test',
      productId: undefined,
      query: {}
    };
    const state = {
      inventory: {
        subscriptionsInventory: {
          fulfilled: true,
          metaId: undefined,
          data: { [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_DATA]: [] }
        }
      }
    };

    expect(subscriptionsListSelectors.subscriptionsList(state, props)).toMatchSnapshot('no product id error');
  });

  it('should handle pending state on a product ID', () => {
    const props = {
      viewId: 'test',
      productId: 'Lorem Ipsum ID pending state'
    };
    const state = {
      inventory: {
        subscriptionsInventory: {
          'Lorem Ipsum ID pending state': {
            pending: true,
            metaId: 'Lorem Ipsum ID pending state',
            data: { [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_DATA]: [] }
          }
        }
      }
    };

    expect(subscriptionsListSelectors.subscriptionsList(state, props)).toMatchSnapshot('pending');
  });

  it('should populate data on a product ID when the api response is missing expected properties', () => {
    const props = {
      viewId: 'test',
      productId: 'Lorem Ipsum missing expected properties',
      query: {
        [rhsmApiTypes.RHSM_API_QUERY_TYPES.SLA]: rhsmApiTypes.RHSM_API_QUERY_SLA_TYPES.PREMIUM
      }
    };
    const state = {
      inventory: {
        subscriptionsInventory: {
          'Lorem Ipsum missing expected properties': {
            fulfilled: true,
            metaId: 'Lorem Ipsum missing expected properties',
            data: {
              [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_DATA]: [
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.PHYSICAL_CAPACITY]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.VIRTUAL_CAPACITY]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.TOTAL_CAPACITY]: 3
                },
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.PHYSICAL_CAPACITY]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.VIRTUAL_CAPACITY]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.TOTAL_CAPACITY]: 3
                }
              ]
            }
          }
        }
      }
    };

    expect(subscriptionsListSelectors.subscriptionsList(state, props)).toMatchSnapshot(
      'data populated, missing properties'
    );
  });

  it('should map a fulfilled product ID response to an aggregated output', () => {
    const props = {
      viewId: 'test',
      productId: 'Lorem Ipsum fulfilled aggregated output',
      query: {
        [rhsmApiTypes.RHSM_API_QUERY_TYPES.SLA]: rhsmApiTypes.RHSM_API_QUERY_SLA_TYPES.PREMIUM
      }
    };
    const state = {
      inventory: {
        subscriptionsInventory: {
          'Lorem Ipsum fulfilled aggregated output': {
            fulfilled: true,
            metaId: 'Lorem Ipsum fulfilled aggregated output',
            data: {
              [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_DATA]: [
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.PHYSICAL_CAPACITY]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.VIRTUAL_CAPACITY]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.TOTAL_CAPACITY]: 3
                },
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.PHYSICAL_CAPACITY]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.VIRTUAL_CAPACITY]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.TOTAL_CAPACITY]: 3
                }
              ]
            }
          }
        }
      }
    };

    expect(subscriptionsListSelectors.subscriptionsList(state, props)).toMatchSnapshot('fulfilled');
  });

  it('should populate data from the in memory cache', () => {
    const props = {
      viewId: 'cache-test',
      productId: 'Lorem Ipsum ID cached',
      query: {
        [rhsmApiTypes.RHSM_API_QUERY_TYPES.SLA]: rhsmApiTypes.RHSM_API_QUERY_SLA_TYPES.PREMIUM
      }
    };
    const stateInitialFulfilled = {
      inventory: {
        subscriptionsInventory: {
          'Lorem Ipsum ID cached': {
            fulfilled: true,
            metaId: 'Lorem Ipsum ID cached',
            data: {
              [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_DATA]: [
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.PHYSICAL_CAPACITY]: 2,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.VIRTUAL_CAPACITY]: 1,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.TOTAL_CAPACITY]: 3
                }
              ]
            }
          }
        }
      }
    };

    expect(subscriptionsListSelectors.subscriptionsList(stateInitialFulfilled, props)).toMatchSnapshot(
      'cached data: initial fulfilled'
    );

    const statePending = {
      inventory: {
        subscriptionsInventory: {
          'Lorem Ipsum ID cached': {
            ...stateInitialFulfilled.inventory.subscriptionsInventory['Lorem Ipsum ID cached'],
            pending: true
          }
        }
      }
    };

    expect(subscriptionsListSelectors.subscriptionsList(statePending, props)).toMatchSnapshot(
      'cached data: cache used and pending'
    );

    const stateFulfilled = {
      inventory: {
        subscriptionsInventory: {
          'Lorem Ipsum ID cached': {
            ...stateInitialFulfilled.inventory.subscriptionsInventory['Lorem Ipsum ID cached'],
            fulfilled: true,
            data: {
              [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_DATA]: [
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.PHYSICAL_CAPACITY]: 3,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.VIRTUAL_CAPACITY]: 3,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.TOTAL_CAPACITY]: 6
                }
              ]
            }
          }
        }
      }
    };

    expect(subscriptionsListSelectors.subscriptionsList(stateFulfilled, props)).toMatchSnapshot(
      'cached data: updated and fulfilled'
    );

    const stateCancelled = {
      inventory: {
        subscriptionsInventory: {
          'Lorem Ipsum ID cached': {
            ...stateInitialFulfilled.inventory.subscriptionsInventory['Lorem Ipsum ID cached'],
            cancelled: true,
            fulfilled: false
          }
        }
      }
    };

    expect(subscriptionsListSelectors.subscriptionsList(stateCancelled, props)).toMatchSnapshot(
      'cached data: ERROR, cancelled API call, maintain prior response'
    );

    const stateFulfilledQueryUpdated = {
      inventory: {
        subscriptionsInventory: {
          'Lorem Ipsum ID cached': {
            ...stateInitialFulfilled.inventory.subscriptionsInventory['Lorem Ipsum ID cached'],
            fulfilled: true,
            data: {
              [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_DATA]: [
                {
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.PHYSICAL_CAPACITY]: 5,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.VIRTUAL_CAPACITY]: 5,
                  [rhsmApiTypes.RHSM_API_RESPONSE_INVENTORY_SUBSCRIPTIONS_DATA_TYPES.TOTAL_CAPACITY]: 10
                }
              ]
            }
          }
        }
      },
      view: {
        [rhsmApiTypes.RHSM_API_QUERY_TYPES.SLA]: rhsmApiTypes.RHSM_API_QUERY_SLA_TYPES.PREMIUM
      }
    };

    expect(subscriptionsListSelectors.subscriptionsList(stateFulfilledQueryUpdated, props)).toMatchSnapshot(
      'cached data: query updated and fulfilled'
    );
  });
});
