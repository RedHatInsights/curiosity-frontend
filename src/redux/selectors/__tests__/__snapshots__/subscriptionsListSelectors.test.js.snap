// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionsListSelectors should handle pending state on a product ID: pending 1`] = `
Object {
  "error": false,
  "fulfilled": false,
  "itemCount": 0,
  "listData": Array [],
  "pending": true,
  "query": Object {},
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should map a fulfilled product ID response to an aggregated output: fulfilled 1`] = `
Object {
  "error": false,
  "fulfilled": true,
  "itemCount": 0,
  "listData": Array [
    Object {
      "physicalCapacity": 2,
      "productName": null,
      "serviceLevel": null,
      "sku": null,
      "subscriptionNumbers": null,
      "totalCapacity": 3,
      "uom": null,
      "upcomingEventDate": null,
      "upcomingEventType": null,
      "usage": null,
      "virtualCapacity": 1,
    },
    Object {
      "physicalCapacity": 2,
      "productName": null,
      "serviceLevel": null,
      "sku": null,
      "subscriptionNumbers": null,
      "totalCapacity": 3,
      "uom": null,
      "upcomingEventDate": null,
      "upcomingEventType": null,
      "usage": null,
      "virtualCapacity": 1,
    },
  ],
  "pending": false,
  "query": Object {
    "sla": "premium",
  },
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should pass minimal data on a product ID without a product ID provided: no product id error 1`] = `
Object {
  "error": false,
  "fulfilled": false,
  "itemCount": 0,
  "listData": Array [],
  "pending": false,
  "query": Object {},
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should pass minimal data on missing a reducer response: missing reducer error 1`] = `
Object {
  "error": false,
  "fulfilled": false,
  "itemCount": 0,
  "listData": Array [],
  "pending": false,
  "query": Object {},
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should populate data from the in memory cache: cached data: ERROR, cancelled API call, maintain prior response 1`] = `
Object {
  "error": false,
  "fulfilled": true,
  "itemCount": 0,
  "listData": Array [
    Object {
      "physicalCapacity": 3,
      "productName": null,
      "serviceLevel": null,
      "sku": null,
      "subscriptionNumbers": null,
      "totalCapacity": 6,
      "uom": null,
      "upcomingEventDate": null,
      "upcomingEventType": null,
      "usage": null,
      "virtualCapacity": 3,
    },
  ],
  "pending": false,
  "query": Object {
    "sla": "premium",
  },
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should populate data from the in memory cache: cached data: cache used and pending 1`] = `
Object {
  "error": false,
  "fulfilled": true,
  "itemCount": 0,
  "listData": Array [
    Object {
      "physicalCapacity": 2,
      "productName": null,
      "serviceLevel": null,
      "sku": null,
      "subscriptionNumbers": null,
      "totalCapacity": 3,
      "uom": null,
      "upcomingEventDate": null,
      "upcomingEventType": null,
      "usage": null,
      "virtualCapacity": 1,
    },
  ],
  "pending": false,
  "query": Object {
    "sla": "premium",
  },
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should populate data from the in memory cache: cached data: initial fulfilled 1`] = `
Object {
  "error": false,
  "fulfilled": true,
  "itemCount": 0,
  "listData": Array [
    Object {
      "physicalCapacity": 2,
      "productName": null,
      "serviceLevel": null,
      "sku": null,
      "subscriptionNumbers": null,
      "totalCapacity": 3,
      "uom": null,
      "upcomingEventDate": null,
      "upcomingEventType": null,
      "usage": null,
      "virtualCapacity": 1,
    },
  ],
  "pending": false,
  "query": Object {
    "sla": "premium",
  },
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should populate data from the in memory cache: cached data: query updated and fulfilled 1`] = `
Object {
  "error": false,
  "fulfilled": true,
  "itemCount": 0,
  "listData": Array [
    Object {
      "physicalCapacity": 5,
      "productName": null,
      "serviceLevel": null,
      "sku": null,
      "subscriptionNumbers": null,
      "totalCapacity": 10,
      "uom": null,
      "upcomingEventDate": null,
      "upcomingEventType": null,
      "usage": null,
      "virtualCapacity": 5,
    },
  ],
  "pending": false,
  "query": Object {
    "sla": "premium",
  },
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should populate data from the in memory cache: cached data: updated and fulfilled 1`] = `
Object {
  "error": false,
  "fulfilled": true,
  "itemCount": 0,
  "listData": Array [
    Object {
      "physicalCapacity": 3,
      "productName": null,
      "serviceLevel": null,
      "sku": null,
      "subscriptionNumbers": null,
      "totalCapacity": 6,
      "uom": null,
      "upcomingEventDate": null,
      "upcomingEventType": null,
      "usage": null,
      "virtualCapacity": 3,
    },
  ],
  "pending": false,
  "query": Object {
    "sla": "premium",
  },
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should populate data on a product ID when the api response is missing expected properties: data populated, missing properties 1`] = `
Object {
  "error": false,
  "fulfilled": true,
  "itemCount": 0,
  "listData": Array [
    Object {
      "physicalCapacity": 2,
      "productName": null,
      "serviceLevel": null,
      "sku": null,
      "subscriptionNumbers": null,
      "totalCapacity": 3,
      "uom": null,
      "upcomingEventDate": null,
      "upcomingEventType": null,
      "usage": null,
      "virtualCapacity": 1,
    },
    Object {
      "physicalCapacity": 2,
      "productName": null,
      "serviceLevel": null,
      "sku": null,
      "subscriptionNumbers": null,
      "totalCapacity": 3,
      "uom": null,
      "upcomingEventDate": null,
      "upcomingEventType": null,
      "usage": null,
      "virtualCapacity": 1,
    },
  ],
  "pending": false,
  "query": Object {
    "sla": "premium",
  },
  "status": undefined,
}
`;

exports[`SubscriptionsListSelectors should return specific selectors: selectors 1`] = `
Object {
  "makeSubscriptionsList": [Function],
  "subscriptionsList": [Function],
}
`;
